integreat_cms/deepl_api/utils.py

```diff
@@ -111,7 +121,12 @@ def deepl_translation(self, request, content_objects, language_slug, form_class)
             )
             # Get target language
             target_language = region.get_language_or_404(language_slug)
-            source_language = region.get_source_language(language_slug)
+
+            source_language = (
+                region.get_language_or_404(source_language_slug)
+                if source_language_slug
+                else region.get_source_language(language_slug)
+            )
```

Comment:
Ahh, this is another one of those remnants of an earlier iteration on this. I was originally always passing the region's root language here, assuming that it would be preferable to avoid "machine translation chains" like DE->EN->FR->ES. Since it's been decided that only one translation step (direct children) should be taken, you are right, this is now superfluous.